### ISSUE TITLE
Use a real JS framework for frontend

### ISSUE THREAD

**Issue by wlonk** [wlonk/wheretofind.me#37]
The profile-editing (CRUD identities, rearrange them, possibly CRUD aliases) and the profile star/unstar logic has gotten to the point where we don't just wanna be slamming it out in jQuery, but we want a bit of a real framework with reusable components and support for frontend tests (see #26).

**Comment by ojacobson** [comment]
If you leave this to me to do, I'll do it in React (plus Redux), probably. I know your preference would be for Ember, but Ember and I have never seen eye-to-eye and it wouldn't be what I'd pick in a vacuum.

Having said that, I don't think that's a decision I _can_ make, and I'm certainly not going to refuse to work on this if it's ember or anything like that. (That preference is reserved for Angular ;) )

**Comment by wlonk** [comment]
@ojacobson actually, my preference a few years ago would have been Ember, but lately React/Redux makes a lot more sense to me. My only concern would be licensing, but that's a concern with the entire FOSS tower this teeters on.

**Comment by wlonk** [comment]
So, @khaledallen volunteered to do some frontend work on this, too, when he's got time. If you and he agree on React/Redux, then I think we have consensus among all the folks likely to do it, and should move ahead with that.

**Comment by ojacobson** [comment]
Vue may also be an option. More of a learning exercise, though. @khaledallen What are your thoughts?

**Comment by khaledallen** [comment]
I mainly have experience with Ember as a framework (or making my own), but all that I've heard of Vue addresses the things I didn't like about Ember. I have no experience with React, but it has never particularly appealed to me. Vue is something I've been interested to learn, and it seems like it allows for incremental adoption as we learn it and as we want to incorporate more of it.

tl;dr since I'd have to learn both react and vue from scratch, personally I would prefer vue.

**Comment by ojacobson** [comment]
I'm going to spend some time tonight Learning Vue. How involved do you want to be (ranging from "I want to lead this" to "whenever I have time but don't wait on me") in a frontend rework, off the cuff - that is, should I work on something else, and synch up with you, or carry on ahead myself and bring you up to speed when you're available?

**Comment by wlonk** [comment]
I can speak to the preferred way to get Webpack (or other JS build process) output into Django, and it's basically that: add the Heroku node buildpack, so that it'll run your build process on deploy, and then have it output to a location in Django's `STATICFILES_DIRS`.

**Comment by khaledallen** [comment]
@ojacobson I'm happy to take a big role, but not right away. I got burnt out from a series of emotionally intensive back-to-back projects over the last couple of months. I'm not sure what the timeline for this is, but it seems like you want to move pretty quickly on it, so go ahead and push forward. I'll jump in next week when I will have had my fill of rejuvenation.
